src/encoding.rs

```diff
@@ -1,6 +1,45 @@
 use data_encoding::Encoding;
 use data_encoding_macro::new_encoding;
 
+// At a later point (probably when floating point arithmatic in const fn is stable) the below functions can be const
+
+#[cfg(feature = "alloc")]
+/// math comes from here https://github.com/bitcoin/bitcoin/blob/f1e2f2a85962c1664e4e55471061af0eaa798d40/src/base58.cpp
+pub(crate) fn gen_encoded_size(base: usize, input_byte_size: usize) -> usize {
```

Comment:
Nitpick: it took me a while to get what this method does as `gen_` confused me. What do you think of `calc_encoded_size()`?